### PR TITLE
ipc: Fail-stop the node when Bitcoin IPC queue overflows

### PR DESCRIPTION
**Pull request by pradhyum6144**
Issue : #318

## Changes

The existing `panic!` in `PriorityRequestQueue::enqueue` never halted the node: it runs inside `tokio::task::spawn_local`, where panics are caught by Tokio and only surface via `JoinHandle::await`. The processor task died silently while callers blocked on `oneshot` channels, leaving the node alive but unable to satisfy the consensus-critical IPC required by `docs/braidpool_spec.md`.

Replace the silent panic with `std::process::abort()` on Normal/High/Critical overflow so operators see the failure. Low-priority overflow stays a soft drop (warn + continue) since it is non-consensus.

### Updates

- **ipc/client.rs**
  - New `EnqueueOutcome` enum returned by `enqueue`, matched by the processor task: Low → warn + drop, Normal/High/Critical → error + `std::process::abort()`.
  - `is_overloaded()` now checks all four priority queues.
  - 7 new unit tests covering each outcome and priority ordering.

### CODE REVIEW

**Comment by pradhyum6144** [comment]
Flagging before review: #318 has two asks, this PR only covers one.

Covered (requirement 1- "halt + noisy logs"): Normal/High/Critical overflow now emits error! + std::process::abort(). Note: the previous panic! never actually halted the node  it ran inside tokio::task::spawn_local, so Tokio caught it and the processor task died silently while callers blocked on oneshot channels. abort() bypasses that.

Not covered (requirement 2 - "ensure this can never happen"): Scoped out; I believe it's blocked on #283 (cmempoold split) + p2p rate limiting.

Direction question: @krishi-agrawal suggested unbounding Critical + High queues so overflow is unreachable - a different philosophy than this PR (bounded + fail-stop). Happy to pivot:

(a) Land as-is, track (2) against #283.
(b) Also unbound Critical + High here, keep abort() as safety net on Normal/Low.
(c) Unbound Critical + High, drop abort() entirely (krishi's proposal).
Which would you prefer?

**Review by zaidmstrr — CHANGES_REQUESTED** [review]
Concept ACK [db6e8f8](https://github.com/braidpool/braidpool/pull/436/changes/db6e8f88a4ccbd78fbb398118409c768a3d10aab)

I don't like the idea of introducing `EnqueueOutcome` enum this is making code complex without adding much meaning, even though the test verifies overflow detection but not the abort response, so you end up testing the enum shape rather than the behaviour. Although your concern about the `abort()` is correct. The cleaner and simpler approach here is to use `abort()` inside `enqueue()` and don't treat low priority requests differently than critical and high. Because if a queue is full on low priority, then still that means the Bitcoin node is unresponsive.

File: node/src/ipc/client.rs [hunk 1]
```diff
@@ -187,63 +195,52 @@ impl PriorityRequestQueue {
             BitcoinRequest::GetMiningTipInfo { priority, .. } => *priority,
             BitcoinRequest::SubmitSolution { priority, .. } => *priority,
         };
-
-        let result = match priority {
```
Comment (zaidmstrr) [inline node/src/ipc/client.rs:191]: Removing this piece of code is not the intended fix for this issue. I suggest just introducing a simple `abort()` and leaving the old code unchanged.

**Review by zaidmstrr — COMMENTED** [review]
Thanks for working on the things I suggested previously. This now looks way better than before, but there are some more suggestions below you can look at.

File: node/src/ipc/client.rs [hunk 2]
```diff
@@ -191,8 +191,8 @@ impl PriorityRequestQueue {
         let result = match priority {
             RequestPriority::Critical => {
                 if self.critical_queue.len() >= self.max_queue_sizes.critical {
-                    error!("BITCOIN IPC QUEUE FULL - CRITICAL QUEUE EXCEEDED LIMIT");
-                    panic!("FATAL: Cannot communicate with Bitcoin Core. Queue is full.");
+                    error!("FATAL: Bitcoin IPC critical queue full - aborting to prevent consensus desync. See issue #318.");
```
Comment (zaidmstrr) [inline node/src/ipc/client.rs:194]: We don't need to change the previous logs please leave them as they are. And sign your commits.

File: node/src/ipc/client.rs [hunk 3]
```diff
@@ -215,8 +215,8 @@ impl PriorityRequestQueue {
             }
             RequestPriority::Normal => {
                 if self.normal_queue.len() >= self.max_queue_sizes.normal {
-                    error!("BITCOIN IPC QUEUE FULL - NORMAL PRIORITY QUEUE EXCEEDED LIMIT");
-                    panic!("FATAL: Cannot communicate with Bitcoin Core. Queue is full.");
+                    error!("FATAL: Bitcoin IPC normal queue full - aborting to prevent consensus desync. See issue #318.");
```
Comment (Sansh2356) [inline node/src/ipc/client.rs:218]: Why is the issue number listed in running logs not required and imo we can revert to previous log messages only .

**Review by Sansh2356 — CHANGES_REQUESTED** [review]
